pkg/apis/admissionregistration/validation/validation.go

```diff
@@ -514,3 +526,231 @@ func ValidateMutatingWebhookConfigurationUpdate(newC, oldC *admissionregistratio
+				allErrors = append(allErrors, field.InternalError(fldPath.Child("expression"), result.Error))
+			default:
+				allErrors = append(allErrors, field.InternalError(fldPath.Child("expression"), fmt.Errorf("unsupported error type: %w", result.Error)))
+			}
+		}
+	}
+	if len(v.Message) > 0 && len(trimmedMsg) == 0 {
+		allErrors = append(allErrors, field.Invalid(fldPath.Child("message"), v.Message, "message must be non-empty if specified"))
+	} else if hasNewlines(trimmedMsg) {
+		allErrors = append(allErrors, field.Invalid(fldPath.Child("message"), v.Message, "message must not contain line breaks"))
+	} else if hasNewlines(trimmedMsg) && trimmedMsg == "" {
+		allErrors = append(allErrors, field.Required(fldPath.Child("message"), "message must be specified if expression contains line breaks"))
+	}
+	if v.Reason != nil && !supportedValidationPolicyReason.Has(string(*v.Reason)) {
+		allErrors = append(allErrors, field.NotSupported(fldPath.Child("reason"), *v.Reason, supportedValidationPolicyReason.List()))
+	}
+	return allErrors
+}
+
+var newlineMatcher = regexp.MustCompile(`[\n\r]+`) // valid newline chars in CEL grammar
+func hasNewlines(s string) bool {
+	return newlineMatcher.MatchString(s)
+}
+
+// ValidateValidatingAdmissionPolicyBinding validates a ValidatingAdmissionPolicyBinding before create.
+func ValidateValidatingAdmissionPolicyBinding(pb *admissionregistration.ValidatingAdmissionPolicyBinding) field.ErrorList {
+	return validateValidatingAdmissionPolicyBinding(pb)
+}
+
+func validateValidatingAdmissionPolicyBinding(pb *admissionregistration.ValidatingAdmissionPolicyBinding) field.ErrorList {
+	allErrors := genericvalidation.ValidateObjectMeta(&pb.ObjectMeta, false, genericvalidation.NameIsDNSSubdomain, field.NewPath("metadata"))
+	allErrors = append(allErrors, validateValidatingAdmissionPolicyBindingSpec(&pb.Spec, field.NewPath("spec"))...)
+
+	return allErrors
+}
+
+func validateValidatingAdmissionPolicyBindingSpec(spec *admissionregistration.ValidatingAdmissionPolicyBindingSpec, fldPath *field.Path) field.ErrorList {
+	var allErrors field.ErrorList
+
+	if len(spec.PolicyName) == 0 {
+		allErrors = append(allErrors, field.Required(fldPath.Child("policyName"), ""))
+	} else {
+		for _, msg := range genericvalidation.NameIsDNSSubdomain(spec.PolicyName, false) {
+			allErrors = append(allErrors, field.Invalid(fldPath.Child("policyName"), spec.PolicyName, msg))
+		}
+	}
+	allErrors = append(allErrors, validateParamRef(spec.ParamRef, fldPath.Child("paramRef"))...)
+	allErrors = append(allErrors, validateMatchResources(spec.MatchResources, fldPath.Child("matchResouces"))...)
+
+	return allErrors
+}
+
+func validateParamRef(pr *admissionregistration.ParamRef, fldPath *field.Path) field.ErrorList {
+	var allErrors field.ErrorList
+	if pr == nil {
+		return allErrors
+	}
+	for i, msg := range path.ValidatePathSegmentName(pr.Name, false) {
+		allErrors = append(allErrors, field.Invalid(fldPath.Child("name").Index(i), pr.Name, msg))
```

Comment:
`i` is just the index of the error message, not part of the API field path the user specified

```suggestion
		allErrors = append(allErrors, field.Invalid(fldPath.Child("name"), pr.Name, msg))
```